src/web/lib/messagePicker.ts

```diff
@@ -0,0 +1,103 @@
+import { record } from '@root/src/web/browser/ophan/ophan';
+
+export type MaybeFC = React.FC | null;
+type ShowMessage = (meta?: any) => MaybeFC;
+
+export type CanShowResult = {
+	result: boolean;
+	meta?: any;
+};
+
+export type Message = {
+	id: string;
+	show: ShowMessage;
+	canShow: () => Promise<CanShowResult>;
+	timeoutMillis: number | null;
+};
+
+type MessageWithTimeout = Message & {
+	cancelTimeout: () => void;
+};
+
+export type MessageConfig = {
```

Comment:
Is this really `MessageConfig`, or is it something more like `SlotConfig`? It could even be something quite different like `Channel`?

Reply:
I really like `SlotConfig` here. I think that reads nicely - the configuration for the slot includes the possible messages we might show in that slot.